src/deepsparse/transformers/pipelines/zero_shot_text_classification.py

```diff
@@ -0,0 +1,425 @@
+    :param num_cores: number of CPU cores to allocate for inference engine. None
+        specifies all available cores. Default is None
+    :param scheduler: (deepsparse only) kind of scheduler to execute with.
+        Pass None for the default
+    :param input_shapes: list of shapes to set ONNX the inputs to. Pass None
+        to use model as-is. Default is None
+    :param alias: optional name to give this pipeline instance, useful when
+        inferencing with multiple models. Default is None
+    :param sequence_length: sequence length to compile model and tokenizer for.
+        If a list of lengths is provided, then for each length, a model and
+        tokenizer will be compiled capable of handling that sequence length
+        (also known as a bucket). Default is 128
+    :param default_model_name: huggingface transformers model name to use to
+        load a tokenizer and model config when none are provided in the `model_path`.
+        Default is "bert-base-uncased"
+    :param model_scheme: training scheme used to train the model used for zero shot.
+        Currently supported schemes are "nli"
+    :param model_config: config object specific to the model_scheme of this model
+        or a dict of config keyword arguments
+    :param num_sequences: the number of sequences to handle per batch.
+    :param labels: static list of labels to perform text classification with. Can
+        also be provided at inference time
+    :param context: context for engine. If None, then the engine will be initialized
+        with 2 streams to make use of parallel inference of labels
+    """
+
+    def __init__(
+        self,
+        *,
+        model_scheme: str = ModelSchemes.nli.value,
+        model_config: Optional[Union[NliTextClassificationConfig, dict]] = None,
+        num_sequences: int = 1,
+        labels: Optional[List] = None,
+        context: Optional[Context] = None,
+        batch_size: Optional[int] = None,
+        **kwargs,
+    ):
+        # Note: users should use num_sequences argument instead of batch_size
+        if model_scheme not in ModelSchemes.to_list():
+            raise ValueError(
+                f"Unknown model_scheme {model_scheme}. Currently supported model "
+                f"schemes are {ModelSchemes.to_list()}"
+            )
+
+        self._model_scheme = model_scheme
+        self._config = self._parse_config(model_config)
+        self._num_sequences = num_sequences
+        self._labels = self._parse_labels(labels)
+        self._thread_pool = None
+
+        # if dynamic labels
+        if self._labels is None:
+            if context is None:
+                # num_streams is arbitrarily chosen to be any value >= 2
+                context = Context(num_cores=None, num_streams=2)
+                kwargs.update({"context": context})
+
+            self._thread_pool = ThreadPoolExecutor(
+                max_workers=context.num_streams or 2,
```

Comment:
can you leave a TODO or asana task for this `2`? We will want to revisit this value when the elastic scheduler has a form of work-stealing across streams